### PR TITLE
drivers: dac: dacx3608: add broadcast register for synchronized output

### PR DESCRIPTION
**Pull request by bbooher**
The dacx3608 line supports a broadcast register so all configured channels can output a singular value, simultaneously. This drastically reduces I2C overhead when using multi-channel control. An API addition was necessary to support a global broadcast channel number. The API addition does not break the write_value() implementation for other DAC drivers in the repo. This change is based on an out-of-tree driver developed internally to handle this use case.

Alternative to the API change, could be a KConfig option or device tree entry. Also, no support for the Broadcast channel was added to the channel_setup() implementation - this may or may not be confusing. I believe it makes sense to maintain explicit setup calls for each channel intended to be configured.

### CODE REVIEW

**Comment by bbooher** [comment]
Partial fix for: https://github.com/zephyrproject-rtos/zephyr/issues/73090

**Review by martinjaeger — COMMENTED** [review]
Thanks for your PR. Only got two minor comments.

Please be aware that we are currently in feature freeze and need to wait a couple of more days before this could get merged after approval.

File: include/zephyr/drivers/dac.h
Comment (bbooher) [reply include/zephyr/drivers/dac.h:108]: Sure can. I chose here because the broadcast channel ID only applies to the dac_write_value() API and not the dac_channel_setup() API in my implementation. If it is higher in the header file, there could be confusion where it could be used. If you have no similar concerns, I'll move it.

File: drivers/dac/dac_dacx3608.c
Comment (bbooher) [reply drivers/dac/dac_dacx3608.c:133]: I have no real preference here - they should be effectively the same. I can switch the data type if bool is more aligned with the rest of the code base.

It *might* better indicate the intent of the code to switch types, but either is very common in C code bases.

File: include/zephyr/drivers/dac.h
Comment (bbooher) [reply include/zephyr/drivers/dac.h]: I've added a note in the comment that it only applies to dac_write_value() and moved to the top of the file as suggested.

**Comment by martinjaeger** [comment]
@bbooher Please have a look at the failed compliance check.

**Comment by bbooher** [comment]
> @bbooher Please have a look at the failed compliance check.

I believe it was a mismatch in git username and signed-off-by name. I used the --signoff parameter this time and re-pushed.

**Review by martinjaeger — APPROVED** [review]
I agree this feature can be added independent of the discussion in #73090 for synchronizing the outputs using the latch pin.

**Comment by martinjaeger** [comment]
@KwonTae-young Added you as a reviewer as you have contributed another TI DAC driver in the past.

**Comment by bbooher** [comment]
@martinjaeger @KwonTae-young - Bumping this PR.

**Review by jilaypandya — APPROVED** [review]

**Pull request merged**
